pkg/tests/cross-tests/input_cross_test.go

```diff
@@ -217,3 +217,203 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
 		SkipCompareRawConfig: true,
 	})
 }
+
+func TestOptionalSetNotSpecified(t *testing.T) {
+	skipUnlessLinux(t)
+	runCreateInputCheck(t, inputTestCase{
+		Resource: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"f0": {
+					Optional: true,
+					Type:     schema.TypeSet,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"x": {Optional: true, Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+		Config:               tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),
+		SkipCompareRawConfig: true,
+	})
+}
+
+func TestInputsEmptyCollections(t *testing.T) {
+	skipUnlessLinux(t)
+	// TODO[pulumi/pulumi-terraform-bridge#1971]
+	t.Skipf("We don't handle missing collections correctly")
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
+
+	// signifies a block
+	resourceElem := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"x": {Optional: true, Type: schema.TypeString},
+		},
+	}
+
+	// signifies an attribute
+	schemaElem := &schema.Schema{
```

Comment:
This looks like a degenerate case instead of the common case, TypeMap of what? The common case is having something like map of numbers or map of strings, which would specify a further nested elem.

Reply:
Good catch, thank you!